practices/understanding_git.md

```diff
@@ -0,0 +1,105 @@
+
+
+
+## Why Git Is Preferred
+
+- Git is much faster than Subversion.
+- Git's repositories are much smaller than Subversions.
+- Git was designed to be fully distributed from the start, allowing each developer to have full local control.
+- Git branches are simpler and less resource heavy than Subversion's.
+- Git branches carry their entire history.
+- Git provides better auditing of branch and merge events
+- Git's repo file formats are simple, so repair is easy and corruption is rare.
+- Git repository clones act as full repository backups.
+
+## Basics of Using Git
+There are a lot of ways that you can start. For your guidance, I have compiled which I call a noob guide to start with git.🙂
+
+You can start by following these steps:
+
+1. First, download git. For windows, you can download it from [here](https://git-scm.com/downloads). For Linux use this command `sudo apt-get update $ sudo apt-get install git`. For MAC you can download it from [here](https://sourceforge.net/projects/git-osx-installer/files/).
+2. Create an empty repository by using the `git init` command or you can checkout a repository by `git clone /path/to/repository` command while using a remote server you can use `git clone username@host:/path/to/repository`.
+3. Now make the changes in the repository according to your need. (This step will follow the workflow that was discussed above)
+4. The changes made by you can be proposed (add to **INDEX**) using `git add <filename>`.
+5. To commit these changes you can use `git commit -m "Commit message"`.
+6. Now the changes that you have made are in the **head** to send it to the remote repository use `git push origin master`. If you have not cloned the repository you can use `git remote add origin <server>`.
+7. If you want to add a feature to your project but don't want to disturb the original project you can make a branch out of the main project by using `git checkout -b feature_x` command. You can switch to the original project by using the `git checkout master` command. Remember unless you push the branch using `git push origin <branch>` it will not be publicly available.
+
+
+Hurrayy!! you created your first branch commit with knowing what exactly goes in the backend.🎉
+
+## Some Pitfalls of the Git 
+
+- User interface maturity
+- Single repository
+- Access controls
+- Binary Files
+
```

Comment:
From my personal experience, I would also add Not enough understanding of Git commands 😄

Reply:
hahahahaha